---
#title: "Healthcare Micro-Environments Define Multidrug-Resistant Organism Persistence"  
#author: "Brendan J. Kelly, MD, MS"  
bibliography: bib/references.bib
csl: csl/journal-of-clinical-microbiology.csl
output: html_document
css: style_helvetica_double.css
---


```{r setup, echo=FALSE, eval=TRUE, include=FALSE, autodep=TRUE}

knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
library(tidyverse)
library(tidybayes)


```
  
  
## Title  
Healthcare Micro-Environments Define Multidrug-Resistant Organism Persistence  
  
## Abbreviated Title  
Distance-Defined MDRO Hotspots  
  
## Authors  
Brendan J. Kelly, MD, MS (1,2,3); Selamawit Bekele (2,3); Sean Loughrey (2,3); Elizabeth Huang (2,3); Pam Tolomeo, MPH (2,3); Michael Z. David, MD, PhD (1,2,3); Ebbing Lautenbach, MD, MPH, MS (1,2,3); Jennifer H. Han, MD, MS (1,2,3); Matthew J. Ziegler, MD, MS (1,2,3) for the CDC Prevention Epicenters Program  
  
## Affiliations  
1 - Division of Infectious Diseases, Department of Medicine;   
2 - Department of Biostatistics, Epidemiology, and Informatics;  
3 - Center for Clinical Epidemiology and Biostatistics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA    
  
## Corresponding Author Contact  
Brendan J. Kelly, MD, MS - brendank@pennmedicine.upenn.edu  
  
## Authors' Contributions  
BJK - study design, data collection and analysis, manuscript;  
SB - subject enrollment and specimen collection;  
SL - specimen processing and analysis;  
EH - specimen collection and processing;   
PT - subject enrollment and specimen collection, manuscript;  
MZD - study design, manuscript;  
EL - study design, manuscript;  
JHH - study design, manuscript;  
MJZ - study design, data collection and analysis, manuscript;  
    
## Disclosures  
The authors report no relevant disclosures.  
  
## Data Availability  
Data, analysis scripts, and model code are available at github.com/bjklab.    
  
## Keywords  
MDRO, hospital microbiome    
  
### Acknowledgements:  
BJK, MJZ, and EL are supported in part by a CDC Cooperative Agreement FOA#CK16-004 "Epicenters for the Prevention of Healthcare Associated Infections”. BJK is supported by the National Institute for Allergy and Infectious Diseases (K23 AI121485 and L30 AI120149). MJZ is supported by the National Institute for Allergy and Infectious Diseases (K23 AI143925). This research was supported by a CDC Cooperative Agreement FOA#CK16-004-Epicenters for the Prevention of Healthcare Associated Infections. 

### Disclosures:  
JHH’s contributions were made at the listed affiliations, but she is currently employed by, and holds shares in, the GSK group of companies.    
   
  
  
-----
\newpage
<P style = "page-break-before: always">
  
## Manuscript  
  
### Abstract  
  
* __Background__: Multidrug-resistant organisms (MDROs) colonizing the healthcare environment have been shown to contribute to risk for healthcare-associated infections (HAIs), with adverse effects on patient morbidity and mortality. We sought to define how the distance from patient and wastewater sources of bacterial contamination define MDRO contamination and persistence in the healthcare environment.     
  
* __Methods__: We performed a prospective cohort study, enrolling 51 hospital rooms at the time of admitting a patient who had a positive culture for an eligible MDRO within the prior 30 days. We performed systematic sampling and MDRO culture of rooms, as well as 16S rRNA sequencing from environmental samples obtained along an axis of from the patient's bed towards the bathroom and/or in-room sink and toilet.      
  
* __Results__: The probability of detecting resistant Gram-negative organisms, including _Enterobacterales_, _Acinetobacter_, and _Pseudomonas_, increased with distance from the patient. In contrast, _Clostridioides difficile_ and methicillin-resistant _Staphylococcus aureus_ were more likely to be detected in close proximity to the patient. Resistant _Pseudomonas_ and _S. aureus_ were enriched in these hotspots despite broad deposition 16S rRNA gene sequences assigned to the same genera, suggesting modifiable factors that permit the persistence of these MDROs.   
    
* __Conclusions__: MDRO hotspots can be defined by distance from patient and wastewater reservoirs. Evaluating how MDROs are enriched relative to bacterial DNA deposition helps to identify healthcare micro-environments and suggests how targeted environmental cleaning or design approaches could prevent MDRO persistence and reduce infection risk.    
  
  
-----
\newpage
<P style = "page-break-before: always">
  
### Importance  
  
Bacteria that are resistant to antibiotics commonly contaminate the healthcare environment and contribute to infections that occur as a result of healthcare. But the different resistant bacteria may persist in different locations, and so require different strategies for eradication. We used a comprehensive sampling strategy to define the hotspots for resistant bacteria in the healthcare environment and how they relate to other environmental features.  
  
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
### Introduction:    
  
The healthcare environment is a reservoir for multidrug-resistant bacterial organisms (MDROs) that contribute to risk for healthcare-associated infections (HAIs). Prior studies have highlighted specific sites in the healthcare environment as reservoirs for MDROs. Surface contamination with MDROs frequently occurs on high-touch objects (HTOs) and within bathroom and wastewater sites in hospitals [@Buchan2019-mh; @Mody2019-nr; @Shams2016-oq; @Weber2013-bg; @Carling2018-dl; @Johnson2018-ve; @Kizny_Gordon2017-rw; and @Regev-Yochay2018-yd]. In the case of _Clostridioides difficile_, this contamination has been shown to contribute directly to infection risk: patients occupying rooms where the previous inhabitant was infected with _C. difficile_ are at increased risk of this infection, and surface contamination has been shown to precede acquisition [@Chen2019-km; @Freedberg2016-bx; @Shaughnessy2011-uf].         
  
The persistence of MDROs and HAI-associated bacteria in the healthcare environment depends on the deposition of bacteria from patients and healthcare workers, environmental features that sustain bacterial growth, and on environmental cleaning practices. Studies of the hospital microbiome have demonstrated bidirectional exchange between environmental surfaces and patient skin, with increased similarity over time [@Lax2017-rw]. However, the hospital environment is not uniform. Because deposition, environmental surface selection, and cleaning practices vary across sites within a single hospital room, the environmental microbiome may likewise vary across close sites. We hypothesized that the competing distance between the source patient and the wastewater reservoirs may serve as an effective means to model bacterial contamination in the hospital room environment.     
  
To define the impact of distance from patient and wastewater sources on the specific MDROs colonizing the healthcare environment, we performed a prospective cohort study with systematic sampling and MDRO culture of rooms occupied by patients from whom an MDRO had been isolated. To further discern the relationship between culturable MDROs and bacterial DNA deposition in the healthcare environment, we complemented our systematic, bacterial culture with 16S rRNA sequencing of the environmental samples. Below we report our findings, which highlight the importance of patient-sources of _C. difficile_ and methicillin-resistant _Staphylococcus aureus_ (MRSA), and the corresponding importance of wastewater sources of multidrug-resistant _Enterobacterales_ and _Pseudomonas_ species.    
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
  
### Materials and Methods:  
  
* __Study Design, Setting, and Population__: We performed a prospective cohort study at the Hospital of the University of Pennsylvania (HUP), an academic medical center in Philadelphia, PA, USA. We enrolled consecutive hospital rooms of newly admitted patients who had a positive clinical or surveillance culture for an eligible MDRO within the prior 30 days. We performed longitudinal, comprehensive bacterial culture to detect MDRO colonization in the environment. Three environmental zones were sampled in each room: (1) near-patient, comprising the patient bed railing, the television remote control, the nurse call button, and the overbed tray table; (2) intermediate distance, comprising the intravenous fluid pump control pad, the supply cart keypad, the in-room computer keyboard and mouse; and (3) sites distant from the patient, comprising the light switch, doorknob, toilet handle and seat, as well as the handles and basin of the bath and/or in-room sinks. A total of 51 subjects were enrolled, and 136 sampling events performed, generating 408 samples. Approval was granted by the University of Pennsylvania Institutional Review Board (IRB protocol #833724).   
  
* __Causal Model__: We hypothesized that the physical distance from patient and wastewater sources of MDROs defines MDRO colonization of the healthcare environment. We further hypothesized that the bacterial DNA deposition in the environment reflects persistent environmental contamination with bacteria and risk for persistent MDRO colonization.     
  
* __Specimen Collection and MDRO Culture__: Upon subject enrollment, environmental surfaces were sampled longitudinally until discharge from the enrolling ICU room, or day 28. Two composite samples were obtained from each in-room area (near, intermediate, and far), with one sponge stick (3M) used for bacterial culture and a flocked swab (Copan) for microbiome analysis. Sampling sites and sampling procedures were standardized with an approximate surface area of less than 50 inches^2^ per site for each sponge stick and 20 linear inches per site for each flocked swab. Sponge stick specimens for bacterial culture were collected in stomacher bags to which phosphate-buffered saline with 0.02% Tween (PBST) was added, with mixture using a mechanical homogenizer (Seward). After centrifugation, the cell pellet was plated on six selective media, including BD MRSA Chromagar, Hardy Chromagar CRE, CHROMagar KPC, Hardy Chromagar ESBL, Thermo SPECTRA VRE, and Anaerobe Systems CCFA-HT. Cultures were incubated aerobically for 18-24 hours at 35-37 C, with the exception of Anaerobe Systems CCFA-HT which was incubated in anaerobic conditions for 18-48 hours. All bacterial colonies with appropriate morphology and chromogenic features were subcultured to blood agar and confirmed by MALDI-TOF with the exception of MRSA. Due to inability to recover isolates after laboratory shutdown during the COVID-19 pandemic, we were unable to assess recovery of one MDRO from three specimens, and we were unable to assess recovery of 6 MDROs from 3 specimens. The presented data and models account for these missing cultures.   
   
* __16S rRNA Gene Sequencing and Bioinformatics__: Parallel surface samples were collected with flocked nylon swabs 16S rRNA gene sequencing and microbiome analysis. Swabs were frozen at -80C prior to DNA extraction and sequencing. DNA extraction was performed using MoBio PowerSoil DNA isolation kit. The V1–V2 hypervariable region of the 16S rRNA gene was amplified using barcoded primers 27F (5′AATGATACGGCGACCACCGAGATCTACACTATGGTAATTGTAGAGTTTGATCCTGGCTCAG-3′) and 338R (5′CAAGCAGAAGACGGCATACGAGATNNNNNNNNNNNNAGTCAGTCAGCCTGCTGCCTCCCGTAGGAGT-3′). Thermocycler conditions were as follows: 5 min at 95 °C, 30*(30 s at 95 °C, 30 s at 56 °C, 90 s at 72 °C), 8 min at 72 °C. 16S rRNA gene sequencing was performed with the Illumina MiSeq platform with 250-bp paired-end reads, permitting high-quality coverage of the 316-bp V1–V2 amplicon. Amplicon sequence variants (ASVs) were identified using DADA2 within the Quantitative Insights Into Microbial Ecology 2 (QIIME2) pipeline [@Callahan2019-ir; @Bolyen2019-sd]. Genus-level taxonomic assignment was performed by comparison to the Greengenes reference database [@McDonald2012-tf], using the naive Bayes classifier implemented in scikit-bio [@Bokulich2018-xf]. To account for the risk of confounding by compositionality, the centered log-ratio transform was applied to genus-level abundances prior to modeling [@Gloor2017-kj].         
  
* __Distance Data Collection__: Distances from the environmental sampling site to the patient bed was recorded using a laser measuring device (Bosch GLM 20). Researchers positioned the laser measure at the head of the bed within the room and recorded distances to the in-room computer workstation, the medication cart, IV pole, and bathroom door. Surfaces in the near group were approximated to be 0.5m from the head of the patient’s bed.      
  
* __Definition of Exposures and Outcomes__: The primary exposure of interest was the distance from the patient (measured towards the wastewater sites). The primary outcome of interest was the detection of a culturable MDRO, including MRSA, _C.difficile_, vancomycin-resistant _Enterococcus_ (VRE), extended-spectrum beta-lactamase (ESBL) producing _Enterobacterales_, carbapenem-resistant _Enterobacterales_, _Acinetobacter_, or _Pseudomonas_. Secondary outcomes of interest included the detection of MDRO genera by 16S rRNA gene sequencing.    
  
* __Statistical Methods__: Data were organized using R statistical software version 3.6.1 [@R_Core_Team2018-xo], and plots generated using the “ggplot2” package [@Wickham2016-eh]. We examined the relationship between distance from the patient (towards the wastewater sites) and the probability of MDRO detection using a Gaussian process model (exponentiated-quadratic kernel) for distance, with models fit separately for each of 11 detected MDROs. The same Gaussian process model evaluated the relationship between distance and the detection of MDRO genera. We also evaluated mixed-effects model incorporating a random effect for subjects to account for clustering of longitudinal data. To understand the relationship between bacterial DNA contamination and risk for MDRO detection, we used a mixed-effects model with random effects for genus and sampling site. Models were fit using Stan Hamiltonian Monte Carlo (HMC) version 2.21, via the "brms" package with default, weakly-informative priors [@Carpenter2017-ec; @Burkner2017-yr]. Prior predictive modeling was performed, and models were fit with 4 chains of 1000 iterations, confirmed with HMC diagnostics (no divergent iterations, Rhat statistic < 1.1 for all parameters, and E-BFMI > 0.2) [@McElreath2016-sf; @Gabry2019-le; @Gelman2020-yu]. We examined parameter distributions at 50%, 80%, and 95% posterior credible intervals to understand the relationship between exposure and outcome variables.  
  
* __Power and Sample Size__: We anticipated that the enrollment of 50 subjects would permit 85% power to detect a difference of 20% in MDRO culture recovery between near and far specimen collection sites, with type 1 error less than 0.05.    
  
* __Availability of Data__: Sequence data has been uploaded to the National Center for Biotechnology Information's Sequence Read Archive (NCBI SRA): PRJNA689547. Data, analysis scripts, and model code are available at https://github.com/bjklab.  
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
  
### Results:  
  
* __Detection of MDROs in the Proximity to Patient-Associated MDROs__: We enrolled 51 patient rooms at the time of admitting a patient who had been recently diagnosed with MDRO colonization or infection by clinical culture. The most common enrolling isolate for patients included in the study was C.difficile (n=16, 31%), followed by ESBL-producing Enterobacterales (ESBL) (n=15, 29%), MRSA (n=13, 25%), VRE (n=7, 14%), and carbapenem-resistant Gram-negative organisms (CRO) (n=4, 8%). Across 51 subjects, we performed 136 sampling events, with each event collecting composite samples from three in-room sites (near, intermediate, and far), and each composite sample plated on six MDRO-selective media. **Table 1** shows the 11 MDROs identified across all enrolled subjects, the total number of times they were recovered from the healthcare environment, and frequency with which they were detected from cultured environmental specimens.        
  
* __Impact of Distance from Patient on MDRO Detection by Bacterial Culture__: We related the probability of MDRO detection by selective culture to the distance from the patient and towards the wastewater sites, using mean distances from each of the three in-room sites sampled at each sampling event. **Figure 1** depicts the impact of distance from patient on each of the 11 MDROs encountered in across our cohort. The probability of detecting ESBL and CR Gram-negative organisms, including _Enterobacterales_, _Acinetobacter_, and _Pseudomonas_, increased with distance from the patient. In contrast, _Clostridioides difficile_ and MRSA were more likely to be detected in close proximity to the patient, and less likely to be detected in the wastewater sites. Of note, VRE was distributed more evenly across all areas. Given the patterns of MDRO detection observed at the genus level, we also examined the relationship between aggregate Gram-negative or Gram-positive MDROs and the distance from patient (**Supplemental Figure 1**). This analysis confirmed the substantial increase in Gram-negative MDRO detection with increasing distance from the patient (and towards the wastewater site), with the opposite pattern observed among Gram-positive MDROs.      
  
* __Impact of Distance from Patient on Bacterial DNA Deposition__: We were able to complement comprehensive, culture-based surveillance with 16S rRNA gene sequencing to detect bacterial DNA deposition in the healthcare environment in a subset of 41 specimens from 16 of our cohort's subjects. In **Figure 2**, we compare the impact of distance from patient on the bacterial DNA reads assigned to the 11 genera from which MDROs were cultured. We found that _Enterobaceriaceae_, including _Citrobacter_, _Enterobacter_, _Klebsiella_, and _Raoultella_, exhibited a similar pattern as did bacterial cultures, increasing with distance from the patient and in proximity to the wastewater sites. In contrast, bacterial DNA assigned to _Acinetobacter_ and _Pseudomonas_ was distributed more evenly across all areas, though bacterial culture was more likely to detect these genera in the wastewater sites. Bacterial DNA reads assigned to _Staphylococcus_ were also distributed evenly across all areas, though bacterial culture was more likely to detect MRSA in close proximity to patients.           
  
* __Relationship Between Bacterial DNA Deposition and MDRO Detection in the Healthcare Environment__: To evaluate the relationship between bacterial DNA contamination detected by untargeted 16S rRNA gene sequencing and environmental MDRO risk, we performed mixed-effects regression to determine the association between genus-level bacterial DNA abundance and the probability of MDRO detection across all observed MDRO genera and sampling sites. **Figure 3** depicts the odds ratio of MDRO detection associated with increasing 16S rRNA gene abundance. A positive association was observed across multiple genera and sites, but the only association that achieved high posterior certainty was _Pseudomonas_ in the far-from-patient site (OR 1.47, 95%CI 1.09 - 1.99).      
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
  
### Discussion:  
  
We sought to define how bacterial deposition and persistence in the healthcare environment is affected by proximity to patient and wastewater sources, and we found clear distinctions between near-patient and far-from-patient/near-wastewater sites in patient rooms. The probability of detecting ESBL-producing and CR Gram-negative organisms, including _Enterobacterales_, _Acinetobacter_, and _Pseudomonas_, increased with distance from the patient. In contrast, _C. difficile_ and MRSA were more likely to be detected in close proximity to the patient, and less likely to be detected in the wastewater sites. In some cases, the pattern of culturable MDROs matched bacterial DNA deposition: _Enterobacterales_ detection increased with distance from the patient and in proximity to the wastewater sites. But in other cases, the discrepancy between bacterial DNA detection and culturable MDROs suggested environmental selection: though bacterial DNA assigned to _Pseudomonas_ and _Staphylococcus_ was distributed evenly across all areas, bacterial culture identified distinct hotspots for CR _Pseudomonas_ and MRSA, far from and near patients, respectively. Of note, both bacterial DNA assigned to _Enterococcus_ and culturable VRE were detected across distances, suggesting that bacterial deposition from stool is extensive. The decreasing probability of _C. difficile_ and decreasing _Clostridium_ DNA deposition as sampling became closer to the wastewater sites may be attributable to more effective cleaning practices specific to these pathogens within wastewater sites.  
  
Because deposition of bacteria from patients and healthcare workers, environmental features that sustain bacterial growth, and environmental cleaning practices vary within patient rooms, we identified micro-environments as we sampled along an axis from the patient's bed towards the wastewater sites. By simultaneously measuring viable MDROs by selective bacterial culture and bacterial DNA deposition by 16S rRNA gene sequencing, and by incorporating distance from patient and wastewater reservoirs, we defined hotspots specific to each MDRO and linked those hotspots to specific features of the healthcare environment. Our approach yields new insight into the contamination of surfaces within the hospital rooms and may be used to guide future studies focused on preventing transmission of MDROs through targeted environmental cleaning approaches.  
  
Our study has several limitations. First, sampling sites were divided into three composite in-room categories rather than multiple individual specimens. While this limits the specificity of surface-specific contamination, this approach decreased the resources needed for sampling a greater number of surfaces in total. Second, bacterial culture was limited to only multidrug-resistant pathogens and _C. difficile_. This approach was taken to limit detection to pathogens of greatest clinical significance within healthcare settings, but it is possible that culturable, non-MDRO bacteria persist in different distributions. Third, culture was non-quantitative to limit complexity of specimen processing. These limitations decrease the specificity of site-specific features, but should not reduce the effectiveness of this methodology for modeling contamination of the healthcare space. Finally, our study was restricted to a single, academic medical center. We expect that the environmental MDROs observed would differ at other medical facilities, according to local prevalence. But we believe that modeling MDRO deposition and persistence using our distance-based approach will yield similar insights.     
  
In conclusion, using distance modeling between the patient bed and wastewater sites is an effective method to identify organism-specific hot spots for bacterial contamination. While further studies are needed to validate this method, this technique may be useful in identifying a patient-specific approach to room design and environmental cleaning to prevent persistence of contamination and transmission of pathogens to subsequent room occupants.  
      
  
  
  
  
-----
\newpage
<P style = "page-break-before: always">
  
  
### Tables:  
  
#### Table 1:   
  
```{r table1, eval=TRUE, autodep=TRUE, error=FALSE, message=FALSE, warning=FALSE, fig.align='center'}

read_csv("./tabs/iceman_cx_data_covid_correction.csv") %>%
  identity() -> iceman_corrected

iceman_corrected %>%
  mutate(genus_label = case_when(genus == "Acinetobacter" ~ "<i>Acinetobacter</i> (CR)",
                                 genus == "Citrobacter" ~ "<i>Citrobacter</i> (ESBL/CR)",
                                 genus == "Clostridioides" ~ "<i>Clostridioides difficile</i>",
                                 genus == "Enterobacter" ~ "<i>Enterobacter</i> (ESBL/CR)",
                                 genus == "Enterococcus" ~ "<i>Enterococcus</i> (VRE)",
                                 genus == "Escherichia" ~ "<i>Escherichia</i> (ESBL/CR)",
                                 genus == "Klebsiella" ~ "<i>Klebsiella</i> (ESBL/CR)",
                                 genus == "Pseudomonas" ~ "<i>Pseudomonas</i> (CR)",
                                 genus == "Raoultella" ~ "<i>Raoultella</i> (ESBL/CR)",
                                 genus == "Staphylococcus" ~ "<i>Staphylococcus</i> (MRSA)",
                                 genus == "Stenotrophomonas" ~ "<i>Stenotrophomonas</i> (ESBL/CR)")) %>%
  group_by(genus_label) %>%
  summarise(positive_cultures = sum(cx_positive, na.rm = TRUE),
            percent_positive = positive_cultures / n(),
            ) %>%
  ungroup() %>%
  gt::gt() %>%
  gt::cols_label(genus_label = "MDRO", positive_cultures = gt::html("Total Positive<br>Cultures"), percent_positive = "Percent Positive") %>%
  gt::fmt_percent(columns = 3) %>%
  gt::fmt_markdown(columns = 1) %>%
  gt::tab_options(table_body.hlines.color = "black",
                  column_labels.border.bottom.color = "black",
                  column_labels.border.top.color = "black",
                  table_body.border.bottom.color = "black",
                  table_body.border.top.color = "black") -> tab1

tab1

```
  
  
__Table 1: Recovery of MDROs from the hospital environment.__ A comprehensive environmental sampling and selective culture strategy captured 11 MDROs. For each MDRO, the table presents the total number of times they were recovered from the healthcare environment, and the frequency with which they were detected from cultured environmental specimens.  
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
### Figures:  
  
#### Figure 1:  
  
  
```{r fig1, eval=TRUE, autodep=TRUE, fig.align='center', out.width='90%'}

knitr::include_graphics(path = "./figs/p_cx_gp.svg")


```
  
  
__Figure 1: Impact of distance from the patient on MDRO detection in the healthcare environment.__ Distance from the head of the patient's bed and towards the wastewater sites is shown on the horizontal axis. The vertical axis depicts the probability of MDRO detection by selective culture for each of 11 detected MDROs according to a logistic regression model incorporating a Gaussian process for distance. The black line shows the best estimate, and shades of red indicate 50%, 80%, and 95% posterior credible intervals. ESBL = extended-spectrum beta-lactamase; CR = carbapenem-resistant; VRE = vancomycin-resistant _Enterococcus_; MRSA = methicillin-resistant _S. aureus_.    
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
#### Figure 2:  
  
```{r fig2, eval=TRUE, autodep=TRUE, fig.align='center', out.width='90%'}

knitr::include_graphics(path = "./figs/p_asv_gp.svg")


```
  
  
__Figure 2: Impact of distance from the patient on bacterial DNA deposition in the healthcare environment.__ Distance from the head of the patient's bed and towards the wastewater sites is shown on the horizontal axis. The vertical axis depicts the probability of bacterial genus detection by 16S rRNA gene sequencing for genera corresponding to each of 11 detected MDROs according to a logistic regression model incorporating a Gaussian process for distance. The black line shows the best estimate, and shades of red indicate 50%, 80%, and 95% posterior credible intervals.         
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
  
#### Figure 3:  
  

```{r fig3, eval=TRUE, autodep=TRUE, fig.align='center', out.width='90%'}

knitr::include_graphics(path = "./figs/p_genus_clr_marginal_per_site.svg")


```
  
  
__Figure 3: Association between bacterial DNA deposition and MDRO detection.__ The odds ratio of MDRO detection associated with each unit increase in CLR-transformed 16S rRNA gene sequence abundance for the corresponding genus is shown for each of the 11 MDROs across each of the three environmental sampling sites. The black points, bars, and lines indicate the posterior median, 50%CI, and 95%CI intervals, respectively. The posterior distribution is depicted above the interval with the area supporting increased detection shaded blue, and the area supporting decreased detection shaded gray.  
  
  
-----  
\newpage
<P style = "page-break-before: always">
  
  
### Supplemental Figures:   
  
#### Supplemental Figure 1:  
  
```{r sfig1, eval=TRUE, autodep=TRUE, fig.align='center', out.width='60%'}

knitr::include_graphics(path = "./figs/p_cx_gp_gram_subject.svg")


```
  
  
__Supplemental Figure 1: Impact of distance from the patient on aggregate Gram-negative and Gram-positive MDROs in the healthcare environment.__ Distance from the head of the patient's bed and towards the wastewater sites is shown on the horizontal axis. The vertical axis depicts the probability of MDRO detection by selective culture for Gram-negative or Gram-positive MDROs in aggregate, according to a logistic regression model incorporating a Gaussian process for distance. The black line shows the best estimate, and shades of red indicate 50%, 80%, and 95% posterior credible intervals.       
  
  
-----
\newpage
<P style = "page-break-before: always">
  
  
## References  
  
<div id="refs"></div>
  
  

  
-----
\newpage
<P style = "page-break-before: always">

  
## R Session Information  
  
```{r session_info, eval=TRUE, autodep=TRUE}

sessionInfo()

```
  
  
  

